From 2818d53f39e04265e389c95d4c1e5abcee756722 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 2 Feb 2016 13:59:46 +0200
Subject: [PATCH 0770/1240] fix: comphy: pcie: fix PCIe sequence (mode fixed
 clock)

- Fixed PCIe sequence - update write to mode fixed clock
- Update the delay to be 10 msec
- Add debug print

Change-Id: I8894c8b133ee1a01ad024865bd0d8b88aac2c103
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27152
Reviewed-by: Shadi Ammouri <shadi@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 drivers/phy/comphy_cp110.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/phy/comphy_cp110.c b/drivers/phy/comphy_cp110.c
index 09fa219..3901ca0 100644
--- a/drivers/phy/comphy_cp110.c
+++ b/drivers/phy/comphy_cp110.c
@@ -100,7 +100,7 @@ static int comphy_pcie_power_up(u32 lane, u32 pcie_by4, void __iomem *hpipe_base
 	data = 0x1 << HPIPE_RST_CLK_CTRL_PIPE_RST_OFFSET;
 	/* Set PHY datapath width mode for V0 */
 	mask |= HPIPE_RST_CLK_CTRL_FIXED_PCLK_MASK;
-	data |= 0x0 << HPIPE_RST_CLK_CTRL_FIXED_PCLK_OFFSET;
+	data |= 0x1 << HPIPE_RST_CLK_CTRL_FIXED_PCLK_OFFSET;
 	/* Set Data bus width USB mode for V0 */
 	mask |= HPIPE_RST_CLK_CTRL_PIPE_WIDTH_MASK;
 	data |= 0x0 << HPIPE_RST_CLK_CTRL_PIPE_WIDTH_OFFSET;
@@ -203,12 +203,13 @@ static int comphy_pcie_power_up(u32 lane, u32 pcie_by4, void __iomem *hpipe_base
 		0x0 << HPIPE_RST_CLK_CTRL_PIPE_RST_OFFSET, HPIPE_RST_CLK_CTRL_PIPE_RST_MASK);
 
 	/* wait 5ms - for comphy calibration done */
-	mdelay(5);
+	mdelay(10);
 
 	debug("stage: Check PLL\n");
 	/* Read lane status */
 	data = readl(hpipe_addr + HPIPE_LANE_STATUS0_REG);
 	if ((data & HPIPE_LANE_STATUS0_PCLK_EN_MASK) == 0) {
+		debug("Read from reg = %p - value = 0x%x\n", hpipe_addr + HPIPE_LANE_STATUS0_REG, data);
 		error("HPIPE_LANE_STATUS0_PCLK_EN_MASK is 0\n");
 		ret = 0;
 	}
-- 
1.9.1

